client/layout/guided-tours/positioning.js

```diff
@@ -110,25 +110,85 @@ function validatePlacement( placement, target ) {
 		: placement;
 }
 
-function scrollIntoView( target ) {
-	const targetSlug = target && target.dataset && target.dataset.tipTarget;
-
+export function getScrollDiff( targetSlug, container ) {
+	const target = targetForSlug( targetSlug );
+	const { top, bottom } = target.getBoundingClientRect();
 	if ( targetSlug !== 'themes' ) {
 		return 0;
 	}
-
-	const { top, bottom } = target.getBoundingClientRect();
-
 	if ( bottom + DIALOG_PADDING + DIALOG_HEIGHT <=
 			document.documentElement.clientHeight ) {
 		return 0;
 	}
-
-	const container = query( '#secondary .sidebar' )[ 0 ];
 	const scrollMax = container.scrollHeight -
 		container.clientHeight - container.scrollTop;
-	const y = Math.min( .75 * top, scrollMax );
 
-	scrollTo( { y, container } );
+	return Math.min( .75 * top, scrollMax );
+}
+
+function scrollIntoView( target ) {
+	const targetSlug = target && target.dataset && target.dataset.tipTarget;
+	if ( targetSlug !== 'themes' ) {
+		return 0;
+	}
+
+	const container = query( '#secondary .sidebar' )[ 0 ];
+	const y = getScrollDiff( targetSlug, container );
+
+	scrollTo( { y: y, container: container, duration: 300 } );
 	return y;
 }
+
+export function getScrolledRect( { targetSlug, scrollY } ) {
+	const target = targetForSlug( targetSlug );
+	let rect = target.getBoundingClientRect();
+	const scrolledRect = {
+		top: rect.top - scrollY,
+		bottom: rect.bottom - scrollY,
+		left: rect.left,
+		right: rect.right,
+		height: rect.height,
+		width: rect.width,
+	};
+	return scrolledRect;
+}
+
+export function getOverlayStyle( { rect } ) {
+	const clientWidth = document.documentElement.clientWidth;
+	const clientHeight = document.documentElement.clientHeight;
+	const correctedRect = {
+		top: rect.top,
+		left: rect.left < 0 ? 0 : rect.left,
+		height: rect.height,
+		width: rect.width,
+		right: rect.left < 0 ? rect.right - rect.left : rect.right,
+		bottom: rect.bottom,
+	};
+
+	return {
+		top: {
+			top: '0px',
+			left: '0px',
+			right: '0px',
+			height: correctedRect.top + 'px',
+		},
+		left: {
+			top: correctedRect.top + 'px',
+			left: '0px',
+			width: correctedRect.left + 'px',
+			height: correctedRect.height + 'px',
+		},
+		right: {
+			top: correctedRect.top + 'px',
+			right: '0px',
+			height: correctedRect.height + 'px',
+			width: ( clientWidth - correctedRect.right ) + 'px',
+		},
+		bottom: {
+			bottom: '0px',
+			height: ( clientHeight - correctedRect.bottom ) + 'px',
+			left: '0px',
+			right: '0px',
+		},
+	};
```

Comment:
I think I chose the ones that I found easier to reason about. Plus there's a case to be made for some combinations being more stable under resizing. E.g. for the top div, fixing left, right, and top to 0px but using height instead of bottom makes it more likely for resizes to be handled gracefully.